src/cpp/src/device_config.hpp

```diff
@@ -29,6 +29,20 @@ class DeviceConfig {
         if (m_device == "CPU") {
             auto inference_precision = core.get_property(device, ov::hint::inference_precision);
             m_kv_cache_type = inference_precision == ov::element::bf16 ? ov::element::bf16 : ov::element::f16;
+            // if user sets precision hint, kv cache type should be changed
+            if (plugin_config.find("INFERENCE_PRECISION_HINT") != plugin_config.end()) {
+                const auto& type_name = plugin_config.at("INFERENCE_PRECISION_HINT").as<std::string>();
+                if (type_name == "f32") {
+                    m_kv_cache_type = ov::element::f32;
```

Comment:
Looks like ov bug. It's always supposed to return f32 for floats as far as I know. There's `set_elemnt_type()` to specify a desired type https://github.com/openvinotoolkit/openvino/blob/9cd8ec5455c618427178a0a427ee9107fa83359f/samples/cpp/classification_sample_async/main.cpp#L96

Reply:
for SPR inference precision is bf16

Reply:
My expectation is that inference precision shouldn't influence in/out tensors dtype in the first place.

Reply:
it's KV cache tensors, which we define by ourselves.

Standard model i/o tensors definitely don't depend on inference precision.